components/script/dom/htmlanchorelement.rs

```diff
@@ -574,45 +574,50 @@ impl Activatable for HTMLAnchorElement {
     }
 }
 
-/// <https://html.spec.whatwg.org/multipage/#the-rules-for-choosing-a-browsing-context-given-a-browsing-context-name>
-fn is_current_browsing_context(target: DOMString) -> bool {
-    target.is_empty() || target == "_self"
-}
-
 /// <https://html.spec.whatwg.org/multipage/#following-hyperlinks-2>
 pub fn follow_hyperlink(subject: &Element, hyperlink_suffix: Option<String>, referrer_policy: Option<ReferrerPolicy>) {
-    // Step 1: replace.
-    // Step 2: source browsing context.
-    // Step 3: target browsing context.
-    let target = subject.get_attribute(&ns!(), &local_name!("target"));
+    // Step 1: TODO: If subject cannot navigate, then return.
+    // Step 2.
+    let replace = false;
+
+    let document = document_from_node(subject);
+    let window = document.window();
+
+    // Step 3: source browsing context.
+    let source = window.window_proxy();
```

Comment:
I think we want `document.browsing_context()` here, which should match https://html.spec.whatwg.org/multipage/browsers.html#concept-document-bc better.